src/Runner.Listener/JobDispatcher.cs

```diff
@@ -346,7 +346,38 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, string orc
                 }
 
                 var term = HostContext.GetService<ITerminal>();
-                term.WriteLine($"{DateTime.UtcNow:u}: Running job: {message.JobDisplayName}");
+
+                string workflowName = string.Empty;
+                if (message.Variables.TryGetValue("system.workflowFilePath", out var workflowPath))
+                {
+                    workflowName = workflowPath.Value.Split('/').LastOrDefault().Trim();
+                }
+
+                string repositoryName = string.Empty;
+                if (message.ContextData.TryGetValue("github", out var context))
+                {
+                    var githubContext = context as Pipelines.ContextData.DictionaryContextData;
+                    if (githubContext?.TryGetValue("repository", out var repository) == true)
+                    {
+                        repositoryName = (repository as Pipelines.ContextData.StringContextData)?.Value?.Trim();
+                    }
+                }
+                
+                string additionalInfo = string.Empty;
+                if (!string.IsNullOrEmpty(workflowName))
+                {
+                    additionalInfo += $"(workflow \"{workflowName}\"";
+                    if (!string.IsNullOrEmpty(repositoryName))
+                    {
+                        additionalInfo += $" in repository \"{repositoryName}\")";
+                    }
+                    else
+                    {
+                        additionalInfo += ")";
+                    }
+                }
+                
```

Comment:
`NameValidation.Sanitize` will remove characters '/' from repo name, which I guess isn't very good. I can change the `NameValidation.Sanitize` method so that '/' is valid character, but I guess it can make the problem elsewhere.

Reply:
I can add an optional argument `allowSlashes` (like already existing optional argument `allowHyphens`) to the `NameValidation.Sanitize` method. I think that in this way I will cover all valid characters in the repository name, and I will not change the remaining method `NameValidation.Sanitize` calls.